server/event_logging/event.py

```diff
@@ -0,0 +1,120 @@
+import json
+from enum import Enum
+from app_config import db
+
+from models import LoggedEvent
+
+
+class Event:
+    """
+    Represents an event relevant during an execution of the simulation.
+    This includes events like the start and end of an execution, as well as actions performed by players.
+    """
+
+    class Type(Enum):
+        PERFORMED_ACTION = "performed_action"
+        EXECUTION_STARTED = "execution_started"
+        EXECUTION_FINISHED = "execution_finished"
+        LOCATION_TAKE_FROM = "location_take_from"
+        LOCATION_LEAVE = "location_leave"
+        PATIENT_ARRIVE = "patient_arrive"
+        PATIENT_LEAVE = "patient_leave"
+
+    def __init__(self, execution: int, type: Type, time: int, data):
+        self.execution = execution
+        self.type = type
+        self.time = time
```

Comment:
`self.time` should be an optional param and in case it is undefined it should be evaluated in the constructor

Reply:
IMO the time parameter ist crucial for the log funcionality. Therefore it is required to set a stamp each time you want to create an event log. Otherwise our replay implementation would be buggy.

Reply:
In fact, I was thinking about not using the absolute time, but rather use the relative time (i.e., the runtime of the execution) for this attribute.